lib/util/DiagramWalkerUtil.js

```diff
@@ -95,13 +92,19 @@ export function participatesIn(participant, shape) {
   return getParticipants(shape).includes(participant);
 }
 
+
 /**
- * Stand-in function for flattening a array of depth 1.
- * @param {Array} array
+ * Returns all elements from the given choreos that match the given type.
+ * @param type {String}
+ * @param choreos {Array}
  * @returns {Array}
  */
-export function flat(array) {
-  // Todo: Replace this with the real flat or flatMap when we have an updated
-  // build pipeline which allows polyfills.
-  return array.reduce((acc, value) => acc.concat(value), []);
+export function getAllElementsOfType(type, choreos) {
+  function traverse(node) {
+    const matches = node.flowElements.filter(fe => is(fe, type));
```

Comment:
`flowElements` can be `undefined` when deleting a newly created choreography, making this line fail.